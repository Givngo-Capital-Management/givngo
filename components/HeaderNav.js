import React from 'react';
import { Image, Menu, Form, Input } from 'semantic-ui-react';
import Link from 'next/link';
import logo from '../assets/images/logo.png';
import './HeaderNav.scss';

export default function HeaderNav() {
  return (
    <Menu borderless className="top-menu" fixed="top">
      <Menu.Item header className="logo">
        <Link href="/home">
          <a href="/home">
            <Image src={logo} size="tiny" />
          </a>
        </Link>
      </Menu.Item>

      <Menu.Menu className="nav-container">
        <Menu.Item className="search-input">
          <Form>
            <Form.Field>
              <Input placeholder="Search" size="small" action="Go" />
            </Form.Field>
          </Form>
        </Menu.Item>
      </Menu.Menu>

      <Menu.Menu position="right">
        <Menu.Item name="features">Features</Menu.Item>
        <Menu.Item name="services">Services</Menu.Item>
        <Menu.Item name="donations">Raise Donations</Menu.Item>
        <Menu.Item name="avatar">
          <Image src="http://via.placeholder.com/80x80" avatar />
        </Menu.Item>
      </Menu.Menu>
    </Menu>
  );
}
