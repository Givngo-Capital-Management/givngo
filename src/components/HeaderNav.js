import React from 'react';
import { Image, Menu, Form, Input } from 'semantic-ui-react';
import Link from 'next/link';
import logo from '../../assets/images/logo.png';
import styles from './HeaderNav.module.scss';

export default function HeaderNav() {
  return (
    <Menu borderless className="top-menu" fixed="top">
      <Menu.Item header className={styles.logo}>
        <Link href="/">
          <a href="/">
            <Image src={logo} size="tiny" />
          </a>
        </Link>
      </Menu.Item>

      <Menu.Menu className={styles.container}>
        <Menu.Item className={styles.input}>
          <Form>
            <Form.Field>
              <Input placeholder="Search" size="small" action="Go" />
            </Form.Field>
          </Form>
        </Menu.Item>
      </Menu.Menu>

      <Menu.Menu position="right">
        <Menu.Item name="features">Features</Menu.Item>
        <Menu.Item name="services">Services</Menu.Item>
        <Menu.Item name="donations">Raise Donations</Menu.Item>
        <Menu.Item name="avatar">
          <Image src="http://via.placeholder.com/80x80" avatar />
        </Menu.Item>
      </Menu.Menu>
    </Menu>
  );
}
