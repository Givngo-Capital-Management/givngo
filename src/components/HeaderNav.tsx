import React from 'react';
import { Image, Menu, Form, Input, Button } from 'semantic-ui-react';
import Link from 'next/link';
import styles from './HeaderNav.module.scss';
import firebase from 'firebase/app'
import 'firebase/auth';
import 'firebase/database';

const SignOutButtonBase = (props: any) => (
  <Button color="orange" type="button" onClick={() => firebase.auth().signOut()}>
    Sign Out
  </Button>
);

const SignOutButton = (SignOutButtonBase);

const SignedInComponent = (
  <Menu.Item name="signout">
    <SignOutButton/>
  </Menu.Item>
)

const SignedOutComponent = (
  <Menu.Item name="avatar">
    <a href="/signin">
      Sign In
    </a>
  </Menu.Item>
)

export default function HeaderNav(props: any) {
  const authUser = firebase.auth().currentUser;
  return (
    <Menu borderless className="top-menu" fixed="top">
      <Menu.Item header className={styles.logo}>
        <Link href="/">
          <a href="/">
            <Image src='/assets/logo.png' size="tiny" />
          </a>
        </Link>
      </Menu.Item>

      <Menu.Menu className={styles.container}>
        <Menu.Item className={styles.input}>
          <Form>
            <Form.Field>
              <Input placeholder="Search" size="small" action="Go" />
            </Form.Field>
          </Form>
        </Menu.Item>
      </Menu.Menu>

      <Menu.Menu position="right">
        <Menu.Item name="features">Features</Menu.Item>
        <Menu.Item name="services">Services</Menu.Item>
        <Menu.Item name="donations">Raise Donations</Menu.Item>
        {authUser ? SignedInComponent : SignedOutComponent}
      </Menu.Menu>
    </Menu>
  );
}
