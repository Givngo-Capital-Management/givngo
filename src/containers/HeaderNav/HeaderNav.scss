.ui.menu.top-menu {
  border: none; /*Remove Semantic UI's bottom border*/
  .logo {
    width: 17rem;
  }
  .nav-container {
    flex-grow: 1;
    padding: 0;
  
    .search-input {
      width: 50%;
      margin: auto;
  
      form {
        width: 100%;
      }
    }
  }
  .header-icon {
    color: #a0a0a0;
  }
}