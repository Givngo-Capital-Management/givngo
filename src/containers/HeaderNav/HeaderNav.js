import React from 'react';
import { Image, Menu, Form, Input } from 'semantic-ui-react';
import './HeaderNav.scss';
import { Link } from 'react-router-dom';
import logo from '../../assets/images/logo.png';

function HeaderNav() {
  return (
    <Menu borderless className="top-menu" fixed="top">
      <Menu.Item header className="logo">
        <Link to="/">
          <Image src={logo} size="tiny" />
        </Link>
      </Menu.Item>

      <Menu.Menu className="nav-container">
        <Menu.Item className="search-input">
          <Form>
            <Form.Field>
              <Input placeholder="Search" size="small" action="Go" />
            </Form.Field>
          </Form>
        </Menu.Item>
      </Menu.Menu>

      <Menu.Menu position="right">
        <Menu.Item name="features">Features</Menu.Item>
        <Menu.Item name="services">Services</Menu.Item>
        <Menu.Item name="donations">Raise Donations</Menu.Item>
        <Menu.Item name="avatar">
          <Image src="http://via.placeholder.com/80x80" avatar />
        </Menu.Item>
      </Menu.Menu>
    </Menu>
  );
}

export default HeaderNav;
