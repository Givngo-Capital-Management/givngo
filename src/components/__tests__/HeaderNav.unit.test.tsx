import React from 'react';
import { shallow } from 'enzyme';
import HeaderNav from '../HeaderNav';

jest.mock('firebase/app', () => {
  return {
    auth: jest.fn(() => ({
      currentUser: ""
    }))
  }
});

test('renders HeaderNav', () => {
  const wrapper = shallow(<HeaderNav />);
  expect(wrapper).toMatchSnapshot();
});
