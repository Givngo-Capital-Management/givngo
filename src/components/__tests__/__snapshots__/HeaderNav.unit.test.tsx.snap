// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HeaderNav 1`] = `
<Menu
  borderless={true}
  className="top-menu"
  fixed="top"
>
  <MenuItem
    className="logo"
    header={true}
  >
    <Link
      href="/"
    >
      <a>
        <Image
          as="img"
          size="tiny"
          src="/assets/logo.png"
          ui={true}
        />
      </a>
    </Link>
  </MenuItem>
  <MenuMenu
    className="container"
  >
    <MenuItem
      className="input"
    >
      <Form
        as="form"
      >
        <FormField>
          <Input
            action="Go"
            placeholder="Search"
            size="small"
            type="text"
          />
        </FormField>
      </Form>
    </MenuItem>
  </MenuMenu>
  <MenuMenu
    position="right"
  >
    <MenuItem
      name="features"
    >
      Features
    </MenuItem>
    <MenuItem
      name="services"
    >
      Services
    </MenuItem>
    <MenuItem
      name="donations"
    >
      Raise Donations
    </MenuItem>
    <MenuItem
      name="avatar"
    >
      <Link
        href="/signin"
      >
        <a>
          Signin
        </a>
      </Link>
    </MenuItem>
  </MenuMenu>
</Menu>
`;
